### PR TITLE
test: Add test of _mm_storeu_ps

### PR DESCRIPTION
**Pull request by HowJMay**

### CODE REVIEW

File: tests/impl.cpp [hunk 1]
```diff
@@ -2026,7 +2026,16 @@ result_t test_mm_storel_pi(const SSE2NEONTestImpl &impl, uint32_t i)
 
 result_t test_mm_storeu_ps(const SSE2NEONTestImpl &impl, uint32_t i)
 {
-    return TEST_UNIMPL;
+    float *f = impl.mTestFloatPointer1;
+    float _a[4] = {1.0, 2.0, 3.0, 4.0};
```
Comment (jserv) [inline tests/impl.cpp:2030]: Why did you pick up the sequence?

Comment (HowJMay) [reply tests/impl.cpp:2030]: let me use `impl.mTestFloats` instead

Comment (marktwtn) [reply tests/impl.cpp:2030]: You just need to rewrite it as `float _a[4];`

File: tests/impl.cpp [hunk 2]
```diff
@@ -2026,7 +2026,16 @@ result_t test_mm_storel_pi(const SSE2NEONTestImpl &impl, uint32_t i)
...
+    __m128 a = _mm_load_ps(_a);
+
+    _mm_storeu_ps(f, a);
```
Comment (marktwtn) [inline tests/impl.cpp:2033]: You are storing the value of `a` to the memory address `f`.
`f` can be declared as an uninitialized array.
`a` should contain random value.

Rewrite as
```c
    float *_a = impl.mTestFloatPointer1;
    float f[4];
```

**Review by marktwtn — CHANGES_REQUESTED** [review]

File: tests/impl.cpp [hunk 3]
```diff
@@ -2026,7 +2026,16 @@ result_t test_mm_storel_pi(const SSE2NEONTestImpl &impl, uint32_t i)
...
+    float *_a = impl.mTestFloatPointer1;
+    float f[4];
+    __m128 a = _mm_load_ps(_a);
+
+    _mm_storeu_ps(f, a);
+    ASSERT_RETURN(f[0] == _a[0]);
+    ASSERT_RETURN(f[1] == _a[1]);
+    ASSERT_RETURN(f[2] == _a[2]);
+    ASSERT_RETURN(f[3] == _a[3]);
+    return TEST_SUCCESS;
```
Comment (marktwtn) [inline tests/impl.cpp:2038]: How about rewrite it as `validateFloat(a, f[0], f[1], f[2], f[3]);`?

File: tests/impl.cpp [hunk 4]
```diff
@@ -2026,7 +2026,16 @@ result_t test_mm_storel_pi(const SSE2NEONTestImpl &impl, uint32_t i)
...
-    return TEST_UNIMPL;
+    float *_a = impl.mTestFloatPointer1;
+    float f[4];
+    __m128 a = _mm_load_ps(_a);
+
+    _mm_storeu_ps(f, a);
+    ASSERT_RETURN(f[0] == _a[0]);
+    ASSERT_RETURN(f[1] == _a[1]);
+    ASSERT_RETURN(f[2] == _a[2]);
+    ASSERT_RETURN(f[3] == _a[3]);
```
Comment (marktwtn) [inline tests/impl.cpp:2037]: These shall be removed.

**Review by marktwtn — CHANGES_REQUESTED** [review]

**Review by marktwtn — APPROVED** [review]

**Pull request merged**